From f247b5a05473dd81041b75e0761f7db9f050e81b Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 9 Oct 2024 11:41:53 +0200
Subject: [PATCH 698/841] fix(dt-bindings): bad FLEXGEN configuration of
 pred-division for STM32MP25

The FLEXGEN pre-divider mask (FLEX_PDIV_MASK) is incorrect.
It is not possible to configure it with a div1024 value.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 50bd8ea3b49b7a4011027c061f52ddcb095f65f1 fix(dt-bindings): bad FLEXGEN configuration of pred-division for STM32MP25]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I30a17f5d2e316f4e6ace7c685843b17fd2efd5e4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/410058
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 include/dt-bindings/clock/stm32mp25-clksrc.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/dt-bindings/clock/stm32mp25-clksrc.h b/include/dt-bindings/clock/stm32mp25-clksrc.h
index 44c252403..0fc28822a 100644
--- a/include/dt-bindings/clock/stm32mp25-clksrc.h
+++ b/include/dt-bindings/clock/stm32mp25-clksrc.h
@@ -56,14 +56,14 @@
 #define CLK_NOMUX	0x00000010
 
 /* Flexgen define */
-#define FLEX_ID_SHIFT	13
-#define FLEX_SEL_SHIFT	9
+#define FLEX_ID_SHIFT	20
+#define FLEX_SEL_SHIFT	16
 #define FLEX_PDIV_SHIFT	6
 #define FLEX_FDIV_SHIFT	0
 
-#define FLEX_ID_MASK	GENMASK_32(18, 13)
-#define FLEX_SEL_MASK	GENMASK_32(12, 9)
-#define FLEX_PDIV_MASK	GENMASK_32(8, 6)
+#define FLEX_ID_MASK	GENMASK_32(25, 20)
+#define FLEX_SEL_MASK	GENMASK_32(19, 16)
+#define FLEX_PDIV_MASK	GENMASK_32(15, 6)
 #define FLEX_FDIV_MASK	GENMASK_32(5, 0)
 
 #define DIV_CFG(div_id, div)	((CMD_DIV << CMD_SHIFT) |\
-- 
2.47.2

